mkosi/__init__.py

```diff
@@ -4126,14 +4128,18 @@ def secure_boot_sign(
         return
 
     with mount():
-        for path, _, filenames in os.walk(root / "efi"):
-            for i in filenames:
-                if not i.endswith(".efi") and not i.endswith(".EFI"):
-                    continue
+        if recursive:
+            paths = [Path(item[0]) for item in os.walk(efi_path)]
+        else:
+            paths = [efi_path]
 
-                with complete_step(f"Signing EFI binary {i} in ESP…"):
-                    p = os.path.join(path, i)
+        for path in paths:
+            for f in itertools.chain(path.glob('*.efi'), path.glob('*.EFI')):
```

Comment:
You can do recursive globs with what in bash parlance is called `globstar`, i.e. `**/*.efi`.

Reply:
Should this function always operate recursively or should it still be optional? Based on that I'll update the code.

Reply:
Since the original is always recursive, this should be, too.

Reply:
Done.